### ISSUE TITLE
User settings/preferences

### ISSUE THREAD

**Issue by MrBartusek** [CircuitVerse/CircuitVerse#849]
**Is your feature request related to a problem? Please describe.**
Circiutverse is a growing platform and lack of any way to customize its behaviour by users is starting to make problems. For an example implementation of new emails system or GDPR compliance are impossible due to lack of settings. 

**Describe the solution you'd like**
Create a new settings page on `/users/:id/settings` that will match new UX style
The page should contain settings divided into categories. In my opinion, hard-coding options and adding them to the database is a bad decision. I think options should be generated via a configuration file in JSON for example:
``` JSON
{
	"name" : "fork-button",
	"display_name" : "Hide fork button on project pages",
	"category:" : "Platform"
}
```
That implementation will make settings harder to implement but will benefit in the future. Settings page should be generated via this JSON file. At database, settings can be put into one array cell. The main values like `Username` `Email` `Password` etc. should be hard-coded cause they won't be changed at all. Also, all settings can should be represented as an object.

The settings should be accessible from the user's dropdown

![image](https://user-images.githubusercontent.com/23432278/73377791-ebc58a80-42bf-11ea-96f0-e6f73eb0df37.png)

**Example fields in configuration**

Account

* Username
* Download your information (GDPR)

Email

* Receive Emails (main checkbox)
* Receive Newstsleller emails (checkbox)
* Receive fork notification emails (always, only milestones, off)
* Receive star notification emails (always, only milestones, off)
* Recieve emails when your circuit is featured (checkbox)

Platform

* Might be used in the future

Security

* Change Password
* 2FA
* Sign out of all other sessions
* Delete account (GDPR)

**Describe alternatives you've considered**
There are no other possible implementations at all

**Are you working on this? (Yes/No)**
No

**Comment by shubhankarsharma00** [comment]
We can also add an option to customize the key bindings.

**Comment by YashKumarVerma** [comment]
how should we implement this in the database ?

**Comment by MrBartusek** [comment]
@YashKumarVerma The best implementation, in my opinion, will be making some sort of `array cell` in a database for each user, rather than hard-coding values cause that will make adding new settings not painful at all

**Comment by shubhankarsharma00** [comment]
> Platform
> 
> * Might be used in the future

@MrBartusek I think I am missing something here. Can you elaborate?

**Comment by Ananthavijay** [comment]
I think notifications can also be put under the settings tab.

**Comment by MrBartusek** [comment]
@shubhankarsharma00 done

**Comment by tachyons** [comment]
Good one

**Comment by harshgujrati** [comment]
Can i work on this @nitin10s  @tachyons

**Comment by tachyons** [comment]
Sure

**Comment by MrBartusek** [comment]
Hey @harshgujrati, I was recently thinking about this issue and I think I could work on it and, as I can see you also claimed #963. Both this and 963 are **huge** projects so what about that: you will work on i18n and I shall work on that

**Comment by harshgujrati** [comment]
Okay @MrBartusek  i will make a pr in which i will make settings for user and will include language option after that you can add more settings and after completing i18n we can work together on more settings, is that okay @MrBartusek

**Comment by MrBartusek** [comment]
@harsha-iiiv are you sure you can handle the whole system that I designed here? That quite a lot of work

**Comment by nitin10s** [comment]
It's best you guys divide the work. I agree with @MrBartusek taking up this and @harsha-iiiv working on i18n.

**Comment by MrBartusek** [comment]
@harshgujrati what about that?

**Comment by harshgujrati** [comment]
I was also saying the same that i will implement i18n and you work on settings, in the pr i was only making the setting tab and will add setting column in user database in which we will store the settings in the form of hash which i have done in the assingment of GSoC after that you can add various settings and we will store language preference of the user there only, is that ok @MrBartusek  @nitin10s

**Comment by MrBartusek** [comment]
@harshgujrati I don't think that would be good idea to make one feature in multiple prs, maybe you will work on i18n and add an option to use it after I complete my part, how about that?

**Comment by harshgujrati** [comment]
okay @MrBartusek

**Comment by vedant-jain03** [comment]
@tachyons Can we discuss this idea in the issues team?

**Comment by ayan-biswas0412** [comment]
> @tachyons @ayan-biswas0412 @Shivansh2407 Can we discuss this idea in the issues team?

sure

**Comment by Jadhav-Dhanashri** [comment]
Hello maintainers ,
I'm new to Circuitverse open source contribution & this would be my first issue. I'd like to work on this if it's available & learn from the process. Could you please assign it to me?
Thank you !
